s3/src/main/scala/akka/stream/alpakka/s3/impl/S3Stream.scala

```diff
@@ -238,6 +239,8 @@ private[alpakka] final class S3Stream(settings: S3Settings)(implicit system: Act
       .mapConcat(r => Stream.continually(r))
       .zip(Source.fromIterator(() => Iterator.from(1)))
 
+  val conditionalAppendFl = Flow[ByteString].orElse(Source.single(ByteString.empty))
```

Comment:
the `Fl` suffix stands a bit out of place. Perhaps call it `atLeastOneByteString`? Then the expression below becomes `SplitAfterSize(chunkSize)(atLeastOneByteString)` which reads nicely.